src/components/sources/sourcesContext.js

```diff
@@ -1,115 +1,150 @@
+            count: sourceNames.length
+          }),
+          message: t('toast-notifications.description', {
+            context: ['deleted-source'],
+            name: sourceNames[0],
+            count: sourceNames.length
+          })
+        },
+        {
+          type: reduxTypes.sources.DESELECT_SOURCE,
+          item: sourcesToDelete
+        },
+        {
+          type: reduxTypes.sources.UPDATE_SOURCES
         }
       ]);
 
-      deleteSource(sourceId)(dispatch);
+      setSourcesToDelete(() => []);
     }
-  }, [sourceId, deleteSource, dispatch]);
 
-  useShallowCompareEffect(() => {
-    if (fulfilled) {
+    if (deletedError && sourcesToDelete.length) {
+      const sourceNames = sourcesToDelete.map(source => source[apiTypes.API_RESPONSE_SOURCE_NAME]);
       dispatch([
         {
           type: reduxTypes.toastNotifications.TOAST_ADD,
-          alertType: 'success',
+          alertType: AlertVariant.danger,
           header: t('toast-notifications.title', {
-            context: ['deleted-source']
+            context: ['error']
           }),
           message: t('toast-notifications.description', {
-            context: ['deleted-source'],
-            name: sourceName
+            context: ['deleted-source', 'error'],
+            name: sourceNames[0],
+            count: sourceNames.length,
+            message: errorMessage
           })
         },
         {
-          type: reduxTypes.sources.RESET_DELETE_SOURCE
-        },
-        {
-          type: reduxTypes.view.DESELECT_ITEM,
-          viewType: reduxTypes.view.SOURCES_VIEW,
-          item: sourceToDelete
+          type: reduxTypes.sources.DESELECT_SOURCE,
+          item: sourcesToDelete
         },
         {
           type: reduxTypes.sources.UPDATE_SOURCES
         }
       ]);
-    }
 
-    if (error) {
-      dispatch({
-        type: reduxTypes.toastNotifications.TOAST_ADD,
-        alertType: 'danger',
-        header: t('toast-notifications.title', {
-          context: ['error']
-        }),
-        message
-      });
+      setSourcesToDelete(() => []);
     }
-  }, [error, fulfilled, message, dispatch, sourceName, sourceToDelete]);
+  }, [deletedError, deletedFulfilled, dispatch, errorMessage, sourcesToDelete, t]);
 
-  return source => {
-    dispatch({
-      type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_SHOW,
+  return sources => {
+    const updatedSources = (Array.isArray(sources) && sources) || [sources];
+
+    onConfirmation({
       title: t('form-dialog.confirmation', {
-        context: ['title', 'delete-source']
+        context: ['title', 'delete-source'],
+        count: updatedSources.length
       }),
       heading: t(
         'form-dialog.confirmation',
         {
           context: ['heading', 'delete-source'],
-          name: source[apiTypes.API_RESPONSE_SOURCE_NAME]
+          count: updatedSources.length,
+          name: updatedSources?.[0]?.[apiTypes.API_RESPONSE_CREDENTIAL_NAME]
```

Comment:
required:
- scan for residual creds apiTypes
- replace API_RESPONSE_CREDENTIAL_NAME with the sources